### PR TITLE
Strengthen warning language around the anonymity set

### LINKED ISSUE #19: Comment from Ben Kaduk
> I agree that there's going to be a tradeoff between larger K's privacy benefit and other considerations, but it seems to me that even the framing as "1/K" is misleading.
> When the sites behind a given fronting service are of differingi popularity, the attacker can do better than 1/K at guessing.  If I'm fronting for a site that gets 900k hits/day and ten that get 10 hits/day, the attacker can just guess that all requests are for the popular site and they'll get the right answer 99.9% of the time.  I do see the disclaimer about "idealized deployment" at the start of the paragraph but I think we could do more to highlight just how idealized this portrayal is.

### PR DESCRIPTION
**Pull request by bemasc**
Fixes #19

### CODE REVIEW

**Review by kaduk — APPROVED** [review]
Thank you!

**Comment by taddhar** [comment]
Looks good to me

File: draft-ietf-tls-svcb-ech.md [hunk 1]
```diff
@@ -185,7 +185,7 @@ _dns 3600 IN SVCB  1 @ ech=ABC... alpn=dot,doq,h3 dohpath=/q{?dns}
...
 
-In an idealized deployment, ECH protects the SNI with an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. Accordingly, an attacker who can enumerate this set can always guess the encrypted SNI with probability 1/K, where K is the number of domains in the set. In practice, this probability may be increased via traffic analysis, popularity weighting, and other mechanisms.
+When Encrypted ClientHello is deployed, an attacker still has many ways to infer the SNI with reduced confidence. Even in an idealized deployment, ECH's protection is limited to an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server. An attacker who can enumerate this set can always guess the encrypted SNI with probability at least 1/K, where K is the number of domains in the set. Some attackers may achieve much greater accuracy using traffic analysis, popularity weighting, and other mechanisms.
```
Comment (martinthomson) [inline draft-ietf-tls-svcb-ech.md]:
```suggestion
When Encrypted ClientHello is deployed, there are many ways that an attacker might infer the SNI. Even in an idealized deployment, ECH's protection is limited to an anonymity set consisting of all the ECH-enabled server domains supported by a given client-facing server that share an ECH configuration. An attacker who can enumerate this set can always guess the encrypted SNI with probability at least 1/K, where K is the number of domains in the set. Some attackers may achieve much greater accuracy using traffic analysis, popularity weighting, and other mechanisms.
```

**Review by martinthomson — COMMENTED** [review]
I think that this could be strengthened further.

**Review by enygren — APPROVED** [review]

**Review by kaduk — APPROVED** [review]

**Review by MikeBishop — APPROVED** [review]

**Pull request merged**